lib/assert.js

```diff
@@ -330,6 +331,15 @@ function getErrMessage(message, fn) {
         const { StringDecoder } = require('string_decoder');
         decoder = new StringDecoder('utf8');
       }
+
+      // ESM file prop is a file proto. Convert that to path.
+      // This ensure opensync will not throw ENOENT for ESM files.
+      const fileProtoPrefix = 'file://';
+      if (StringPrototypeStartsWith(filename, fileProtoPrefix)) {
+        const fileUrl = new URL(filename);
+        filename = fileURLToPath(fileUrl);
```

Comment:
```suggestion
        filename = fileURLToPath(filename);
```

Reply:
filename is a string which has file:// prefix. But, it's not an URL object. So, we are parsing to URL first and converting it.

Reply:
`fileURLToPath` already does that for you, there's no point of doing it here

Reply:
Yeah. You are right. I missed the line `if (typeof path === 'string')`

Reply:
modified.